src/module.cpp

```diff
@@ -554,6 +554,15 @@ instruction_ref module::get_parameter(std::string name) const
         return this->end();
 }
 
+void module::rename_parameter(instruction_ref ins, const std::string name)
+{
+    assert(ins->name() == "@param");
+    auto& op      = const_cast<operation&>(ins->get_operator());
+    auto param_op = any_cast<builtin::param>(&op);
```

Comment:
Just do the `any_cast` in the same line: `auto op = any_cast<builtin::param>(ins->get_operator())`.

Reply:
I have made the changes to the `rename_parameter` function and it looks like this now:
```
void module::rename_parameter(instruction_ref ins, const std::string name)
{
    assert(ins->name() == "@param");
    auto op = any_cast<builtin::param>(ins->get_operator());
    op.parameter = name;
    ins->replace(op);
}
```
But I get the following runtime error:
```
[   RUN    ] autocast_fp8_1
terminate called after throwing an instance of 'migraphx::version_2_9_0::exception'
  what():  /code/AMDMIGraphX/AMDMIGraphX/src/include/migraphx/builtin.hpp:78: compute_shape: builtin
Aborted (core dumped)
```
Just wanted to make sure I am not missing any changes that you suggested @pfultz2.

Reply:
The function `get_operator()` does not allow modification to the operation object that it returns (defined as: `const operation& instruction::get_operator() const { return op; }`). I think this is what is causing the issue.

If I have another function that allows the modification, then the tests run fine without any issues. How do you think we should approach this in light of the changes you suggested @pfultz2?

Reply:
We could just use assignment here:

```cpp
    auto op = any_cast<builtin::param>(ins->get_operator());
    op.parameter = name;
    *ins = instruction{op, ins->get_shape(), {}};
```

Reply:
With that change, I see the following error:
```
[   RUN    ] autocast_fp8_1
terminate called after throwing an instance of 'migraphx::version_2_9_0::exception'
  what():  /code/AMDMIGraphX/AMDMIGraphX/src/pass_manager.cpp:56: validate_pass: autocast_fp8_pass
pass produces invalid program at instruction 4: add
Aborted (core dumped)
```

Reply:
Can you push your changes?

Reply:
Sure, done. The last two commits should have the changes you suggested.

Reply:
Theres a use-after-free error here.

Reply:
So this clears the outputs so the `replace_instruction` doesnt do anything. We need to add back the outputs:

```cpp
    auto outputs = ins->outputs();
    *ins = instruction{op, ins->get_shape(), {}};
    for(auto output:outputs)
        ins->add_output(output);
```

Reply:
Ah, I see. I have pushed the commit with this change. I do not see an error with the tests now. Thanks a lot for the help, Paul!